src/main/java/net/snowflake/client/jdbc/SnowflakeChunkDownloader.java

```diff
@@ -562,8 +581,10 @@ private void waitForChunkReady(SnowflakeResultChunk currentChunk) throws Interru
           currentChunk.getDownloadState(),
           retry);
 
-      if (currentChunk.getDownloadState() != DownloadState.FAILURE) {
-        // if the state is not failure, we should keep waiting; otherwise, we skip waiting
+      if (currentChunk.getDownloadState() != DownloadState.NOT_STARTED
```

Comment:
It's now possible we reach waitForChunkReady() function without having started the chunk download if the prefetch chunk download was cancelled. If this is the case, add in logic to start the chunk download.